lib/auth/auth.go

```diff
@@ -245,6 +246,12 @@ func NewServer(cfg *InitConfig, opts ...ServerOption) (*Server, error) {
 	if cfg.UserPreferences == nil {
 		cfg.UserPreferences = local.NewUserPreferencesService(cfg.Backend)
 	}
+	if cfg.UserLoginState == nil {
+		cfg.UserLoginState, err = local.NewUserLoginStateService(cfg.Backend)
```

Comment:
Should this be a memory backend here?

Reply:
I've migrated this to use a memory backend so that it can be quickly cached/looked up elsewhere.